### ISSUE TITLE
Opening Gitter rooms for OSSU-CS cohorts

### ISSUE THREAD

**Issue by bradleygrant** [ossu/computer-science#543]
I am opening this issue on behalf of the members of [OSSU-CS Cohort #23](https://github.com/ossu/cohorts/issues/23).

There has been discussion on the [OSSU Computer Science Gitter room](https://gitter.im/open-source-society/computer-science) about providing an OSSU-sanctioned Gitter room for each of several individual cohorts. The purpose is for cohort members to discuss and share information about the classes they're currently taking, ask questions specific to those courses, check in with their coordinators, and share wins and challenges with their cohort members.

It was suggested that these cohort meeting rooms operate on the main OSSU Gitter page, for several reasons, including but not limited to:

- To provide a central meeting place for all cohorts and all participants within a cohort;
- To bring more visibility to the OSSU cohort system, and encourage other OSSU participants to "go and do likewise" and form their own cohorts;
- For OSSU internal tracking of candidate engagement and retention;
- As a form of social proof that the OSSU curriculum is being used successfully;
- In attempt to achieve a critical mass on the Gitter platform rather than splintering OSSU-related services to Yet Another Platform;
- and other reasons as yet unnamed or unidentified.

This issue is twofold:

1. The community is requesting guidance from OSSU as to whether the goals and reasoning listed herein are worthy and representative of the OSSU mission, and whether OSSU desires and intends to support them. **Key Deliverable:** A decision from OSSU as to whether or not they would like to host cohort rooms on the OSSU Gitter page.
2. If affirmative, we would like to implement cohort rooms on the OSSU Gitter page. **Key Deliverable:** We ask the OSSU Gitter moderator(s) to create a cohort room for Cohort #23.
2. (2a) If negative, we request an opinion from OSSU as to what other OSSU-sanctioned space we may use, or where to share a link to our independently-operated meeting space. 

Thank you.

**Comment by hanjiexi** [comment]
Hi @bradleygrant!

I was talking with a couple OSSU leads about this. The basic answer is yes, if possible **we want cohorts to be situated in the same place as our general community chat**, and it therefore ought to be possible for to create arbitrary rooms for cohorts under the OSSU Gitter. There should probably be only one or two people with this responsibility — cohort coordinators, which currently I know we have @amorriscode.

Unfortunately we can't figure out how!

Both he and I have attempted pressing "Add A Room", and it forces us to select a community with which to associate the room. But the open-source-society community under which our current rooms fall is not listed as an option. And I can find no interface anywhere that would allow me to modify the overall settings for the whole community. There is a very limited settings dialog box for individual rooms, but it doesn't offer any helpful options.

There is a button labeled "Edit Community Permissions", which warned that switching from GitHub-based administrators to Gitter-based administrators is permanent. I decided to try it (I was in the computer-science room), and the only apparent change it made is that... now I'm no longer an administrator in the computer-science room! 😅 And still no community permissions settings. (I'm still an admin in other OSSU rooms, so apparently the only change it made was localized to the CS room.)

To see what a fully-featured community looks like, I created a test community. The settings are really, really limited. Take a look.

Here's what the room settings look like:
![screenshot from 2018-12-06 09-50-54](https://user-images.githubusercontent.com/5695330/49591570-c697ef00-f93c-11e8-8a26-865e6faff18c.png)

Creating a room:
![screenshot from 2018-12-06 09-51-15](https://user-images.githubusercontent.com/5695330/49591589-d8799200-f93c-11e8-9975-d9c3d9e19a95.png)

Community settings:
![screenshot from 2018-12-06 09-51-01](https://user-images.githubusercontent.com/5695330/49591582-cef02a00-f93c-11e8-87ae-65b8b06e4136.png)

So all you can basically do as an admin is create a room, add admins to a room, and add admins to a community. There are no fine-grained permission levels or roles, and there is no way to organize rooms into categories (like a "Cohorts" category), which would probably be very useful as the number of cohorts grows.

Here's something else I tried. I pressed the "create a new community" button and it did allow me to select the ossu organization. I didn't finalize this, as I didn't want to introduce a potentially confusing fork to our Gitter community. I have a hypothesis that the open-source-society Gitter community is in a semi-broken state because the GitHub org it was associated to (github.com/open-source-society) changed its identifier (now it's github.com/ossu). If this is the case, then making a new Gitter community may be the only option to salvage Gitter as our platform.

So that leaves us with three options:
1. Try to fix our current Gitter. This may only be possible from the account of the person who created it in the first place (probably @ericdouglas? but I wasn't able to reach him). Since the association to our GitHub org doesn't seem to be fully functional, someone with sufficient admin privilege would need to try switching it to a Gitter-based permission system. If that doesn't work even for the original creator of the Gitter community, then we'll _know_ that the Gitter community is broken.
2. Just go ahead and get everyone to move to a new, non-broken Gitter community. As with option 1, this comes with all the downsides of Gitter itself — very limited settings for organizing rooms and setting permissions.
3. Start debating the merits of other platforms for OSSU to move to. (In the short-term, perhaps different cohorts can try different ones and report back their impressions.)

Option 1 may not be possible if the original admin is unavailable. I am totally fine with option 2, as long as everyone understands that the Gitter community channel list is going to become a cluttered mess if we ever reach the point of having more than, say, 20 cohorts running at once. I might prefer option 3 if something better is out there.

**Comment by ericdouglas** [comment]
I'd suggest two solutions for this demand:

1. Open an issue on ossu/forum called "Cohort #23", and participants can discuss there topics related to the cohort. For questions about courses, student should open a new issue on the forum, and such student can make a reference to the cohort writing `#number-of-cohort-issue` in the issue so other members of the cohort will see the ping in the cohort issue and can go in that another issue to answer. [Example of successful forum using Github issues](https://github.com/frontendbr/forum/issues)

2. Move from Gitter to Discord. The Vue community is using Discord to host their chat rooms and they have great possibilities to host under the same place chat rooms about different topics. [Link to Vue Land](https://vue-land.js.org/)

![Sections and subsections in the Vue Land](https://user-images.githubusercontent.com/4119931/49685897-959df280-fad3-11e8-8aef-eadbec722bf2.png)
> Sections and subsections in the Vue Land

In fact, I think both solutions are complementary and not excluding. I'd vote to move from Gitter to Discord AND for the specific question raised in this issue, I'd use ossu/forum, but we can easily open a section in Discord called "Cohorts" and put chats inside for each cohort, but it should be more used for informal conversation. In order to ask course related questions, the forum is a better place because it can be accessed for anyone, even if the person is not a member of ossu.

**Comment by hanjiexi** [comment]
Discord was on my mind when I suggested looking at alternatives for semi-real time chat. The experience is indeed very high quality.

Slack and Discord are probably the two that people will suggest most often, each having their own minor inconvenience — the video game focus of Discord, and the focus of Slack not on large, open communities but on private teams. I think [Zulip](https://zulipchat.com/) would be more manageable than Slack for a large community due to its conversation threading model, but it's still ultimately designed for teams, not communities.

Discord allows [GitHub integration](https://support.discordapp.com/hc/en-us/articles/228383668-Intro-to-Webhooks) so it would probably be fine for OSSU, but I don't like making hasty decisions so I want to see if anyone knows of anything that is basically just like Discord, but not video game focused.

**Comment by bradleygrant** [comment]
@ericdouglas @hanjiexi et al

I appreciate all of your suggestions and effort in trying to track down a suitable option. For the time being our cohort coordinator has made a Gitter room for our use, and I think it'll be a great option from our perspective -- and easy enough to invite in anyone else who expresses interest, though perhaps not very easy to find (although, it is posted on the Github cohort issue page).

I'm fine with this as an option for now. Did we want to follow up on any of the other issues that were raised? If not (or not now), I'm comfortable closing this issue.

**Comment by Alaharon123** [comment]
Before you close the issue, really we're just waiting on Eric Douglas to give someone else permission so we can add rooms to the ossu gitter, right?

**Comment by waciumawanjohi** [comment]
1) @bradleygrant Glad to see that your cohort is moving forward and trying out a live chat room for the class. I hope that it leads to greater engagement and success among the learners. Please share back your impressions.

2) @ericdouglas Do you have the ability to make rooms under the OSSU gitter? If so, does Gitter allow you to share that ability with others? Ensuring @hanjiexi and I can create new rooms would enable OSSU to respond to students' requests promptly.

3) Zulip is an interesting find. I think they've identified a major pain point of Slack, it can be hard to follow a conversation. It looks like a good option to have a cohort explore sometime in the future and report back on their experience.

**Comment by bradleygrant** [comment]
This cohort has been established for nearly 8 months and seems to have run its course; because of that I'm closing this issue. Thanks all for your help.
